app/frontend/src/features/map/mapActions.ts

```diff
@@ -0,0 +1,18 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+import { service } from "../../service";
+
+export const onLand = createAsyncThunk(
+  "map/onLand",
+  async ({ lat, lng }: { lat: number; lng: number }) => {
+    const onLand = await service.gis.isOnLand(lat, lng);
+    console.log(`On land: ${onLand}`);
```

Comment:
This is not ready yet, right?

Reply:
Yes, requires fixing up.

Reply:
Do you know how I can get the result back into `MapPage.tsx` that fires this with a `dispatch(...)`?

Reply:
Dispatch is only necessary for redux. You can just call it as a normal async function. If you want to display it on the page, use a state hook near the top of the component `const [isOnLand, setIsOnLand] = useState(false);` then in handleClick do `setIsOnLand(await onLand(...))`. 

Later, if you want loading status and other fancy async stuff for free, you can use react-query (see our current pull requests), or just for testing, you can have another state hook `[isLoading, setIsLoading]`.